src/server/dto/users/userResponse.dto.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * @module server/dto/users/userResponse
+ */
+
+/**
+ * User Response
+ *
+ * Represents the format of user objects returned by the server in response to
+ * API requests for users (i.e: getting the current user, getting a list of all
+ * users etc.)
+ */
+
+export abstract class UserResponse {
```

Comment:
I'm a little concerned with having the same dto defined in two different files -- that seems like a lot of potential for error if we want to change things in the future. Example: say we decide to use `surname` instead of `lastName`, after we've already created multiple `User*` dtos. That's a lot to update for one small change...

What if we define our shared `dto`'s in the common directory, then we could re-export them (or `extend` them if we need to make changes) from the `client`/`server` directories?

Reply:
I know what you mean and I kind of agree. At the same time, I would be wary of doing that for a couple of reasons...

1. There are things we're going to end up with in our server-side DTOs(for example validation with [class-validator](https://github.com/typestack/class-validator)) that don't necessarily make sense on the client side and possibly vice versa...
1. If our API changes field names etc. we would have to do refactoring anyway of UI components, database entities etc. so this would just shift the boundary of the problem...

Reply:
Fixed in 2a86c48d4a7502dbdf281a9f7c621ef773bcddc3

Reply:
...and 2fd2139a17b7f94af62ba72971026473e93c5b03